### PR TITLE
modules: hal_nxp: osa: Remove unused OSA API

### PR DESCRIPTION
**Pull request by Radimli**
- Deleted file fsl_os_abstraction.h is unused and deprecated
- Location of current OSA API available in modules/hal/nxp/mcux/mcux-sdk/components/osa

### CODE REVIEW

**Review by danieldegrasse — COMMENTED** [review]
Thanks for this change- please take a look at your `git` configuration. There are a few issues the compliance script is flagging (you can view these by clicking "Details" on the failed compliance check)

- Commit author does not match the `signed-off` entry: You can update the author of this commit with `git commit --amend --author "Radim.Lipka@nxp.com"`
- Commit body is too long (>75 characters on one line)- just add a line break in the commit body to fix this, editors like vim will do this for you automatically

Edit: we will also need to look at the failing twister tests- these point to places where the OSA API might still be used. It looks like this PR has some USB sample build failures.

**Comment by Radimli** [comment]
> Thanks for this change- please take a look at your `git` configuration. There are a few issues the compliance script is flagging (you can view these by clicking "Details" on the failed compliance check)
> 
> * Commit author does not match the `signed-off` entry: You can update the author of this commit with `git commit --amend --author "Radim.Lipka@nxp.com"`
> * Commit body is too long (>75 characters on one line)- just add a line break in the commit body to fix this, editors like vim will do this for you automatically
> 
> Edit: we will also need to look at the failing twister tests- these point to places where the OSA API might still be used. It looks like this PR has some USB sample build failures.

Hello @danieldegrasse,

yes, cmake in USB MW should be modified. Already discussing it with @MarkWangChinese

**Comment by Radimli** [comment]
@danieldegrasse @MarkWangChinese 
It should be solved in 
https://github.com/zephyrproject-rtos/hal_nxp/pull/433

**Comment by MarkWangChinese** [comment]
Hi @Radimli You can update this PR (changing the west.yml) to contain the https://github.com/zephyrproject-rtos/hal_nxp/pull/433 like: https://github.com/zephyrproject-rtos/zephyr/pull/77973/files#diff-d7f4082837dc3cb270fcf1abd68188216fdadee294e01611b8ca59ddea7fdbb3R201, then let the CI to check it.

File: west.yml
Comment (danieldegrasse) [inline west.yml:201]:
```suggestion
      revision: pull/433/head
```
Don't grab the new SHA directly, that won't be possible until the PR in the HAL merges. Instead, use syntax like I suggested above so west can pull in the contents of the PR

**Comment by danieldegrasse** [comment]
Build tests look good, but you still have a compliance error. If you want to check this locally rather than waiting on CI (and are using Linux) you can run `./scripts/ci/check_compliance.py -c $(git merge-base HEAD main)...HEAD` from your zephyr repo, with this branch checked out.

**Review by danieldegrasse — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
Please address merge conflict when you get a chance

**Review by danieldegrasse — DISMISSED** [review]

**Comment by danieldegrasse** [comment]
@Radimli ping to fix merge conflict

**Review by dleach02 — DISMISSED** [review]

**Comment by dleach02** [comment]
@Radimli we are working to sequence the HAL PRs

**Review by danieldegrasse — DISMISSED** [review]

**Review by dleach02 — DISMISSED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**
